docs/mp/metrics/metrics.adoc

```diff
@@ -579,6 +588,42 @@ curl -H "Accept: application/json"  http://localhost:8080/metrics/application
 ----
 <1> The application has been running for 6 seconds.
 
+[[extension_example]]
+==== Working with Metrics in CDI Extensions
+
+You can work with metrics from your own CDI extension by observing the `RuntimeStart` event.
+
+[source,java]
+.CDI Extension that works correctly with metrics
+----
+import io.helidon.microprofile.cdi.RuntimeStart;
+
+import jakarta.enterprise.event.Observes;
+import jakarta.enterprise.inject.spi.Extension;
+import org.eclipse.microprofile.metrics.Counter;
+import org.eclipse.microprofile.metrics.MetricRegistry;
+
+public class MyExtension implements Extension {
+
+    void startup(@Observes @RuntimeStart Object event,  // <1>
+                 MetricRegistry metricRegistry) {       // <2>
+        metricRegistry.counter("myCounter");         // <3>
+        }
+}
+----
+<1> Declares that your observer method responds to the `RuntimeStart` event. By this time, Helidon has initialized the metrics system.
+<2> Injects a `MetricRegistry` (the application registry by default).
+<3> Uses the injected registry to register a metric (a counter in this case).
+
+[NOTE]
+====
+Helidon does not prevent you from working with metrics earlier than the `RuntimeStart` event, but if you do so Helidon might ignore certain configuration settings that would otherwise control how metrics behaves.
```

Comment:
but, if you do so, then Helidon might ignore

Reply:
Fixed